packages/fxa-settings/src/lib/gql.ts

```diff
@@ -0,0 +1,52 @@
+import { ApolloClient, createHttpLink, from } from '@apollo/client';
+import { setContext } from '@apollo/client/link/context';
+import { onError } from '@apollo/client/link/error';
+import { cache, sessionToken, typeDefs } from './cache';
+
+export function createApolloClient(gqlServerUri: string) {
+  // httpLink makes the actual requests to the server
+  const httpLink = createHttpLink({
+    uri: `${gqlServerUri}/graphql`,
+  });
+
+  // authLink sets the Authentication header on outgoing requests
+  const authLink = setContext((_, { headers }) => {
+    return {
+      headers: {
+        ...headers,
+        Authorization: sessionToken(),
+      },
+    };
+  });
+
+  // errorLink handles error responses from the server
+  const errorLink = onError(({ graphQLErrors, networkError }) => {
+    let reauth = false;
+    if (graphQLErrors) {
+      for (const error of graphQLErrors) {
+        if (error.extensions?.code === 'UNAUTHENTICATED') {
+          reauth = true;
+        }
+      }
+    }
+    if (networkError && 'statusCode' in networkError) {
```

Comment:
Is `networkError && 'statusCode' in networkError` equivalent to `networkError?.statusCode`?

Reply:
in this case unfortunately not. The type definition for `networkError` is a union of a bunch of different types so we need the `&&` and `in` as a type guard since there's not shorter syntax for "not undefined and this (more) specific type"